bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyScriptEngineConfiguration.java

```diff
@@ -294,13 +276,134 @@ synchronized void configureGems(ScriptEngine engine, boolean force) {
+     * 
+     * If the content is empty, Bundler will be skipped.
+     * 
+     * @return the path to the Gemfile or null if no Gemfile is found or configured
+     */
+    public @Nullable String getGemfilePath() {
+        boolean createGemfileFromConfig = false;
+        Path gemfilePath = Path.of(configuration.bundle_gemfile_path);
+
+        if (gemfilePath.equals(Path.of(""))) {
+            gemfilePath = Path.of(DEFAULT_GEMFILE);
+        } else if (!gemfilePath.isAbsolute()) {
+            gemfilePath = Path.of(HOME_PATH_ABS).resolve(gemfilePath);
+        }
+
+        if (gemfilePath.toFile().exists()) {
+            return gemfilePath.toString();
+        }
+
+        String configuredGemfileString = String.join("\n", configuration.bundle_gemfile_content);
+        if (configuredGemfileString.isBlank()) {
+            return null;
+        }
+
+        gemfilePath = Path.of(UI_GEMFILE_PATH);
+        Path gemfileDirectory = gemfilePath.getParent();
+        if (!gemfileDirectory.toFile().exists()) {
+            logger.debug("Creating Gemfile directory {}", gemfileDirectory);
+            gemfileDirectory.toFile().mkdirs();
+        }
+
+        File gemfile = gemfilePath.toFile();
+        try {
+            List<String> headers = List.of( //
+                    "# This Gemfile is auto-generated by the JRuby Scripting add-on", //
+                    "# Do not edit this file directly.", //
+                    "# Edit the JRuby Scripting configuration 'bundle_gemfile_content' instead.", //
+                    "");
+
+            List<String> configuredGemfileContent = new ArrayList<>(headers);
+            configuredGemfileContent.addAll(configuration.bundle_gemfile_content);
+
+            if (gemfile.exists()) {
+                List<String> fileContent = Files.readAllLines(gemfilePath, StandardCharsets.UTF_8);
+                if (fileContent.equals(configuredGemfileContent)) {
+                    logger.debug("Gemfile already exists and is up to date.");
+                    return gemfilePath.toString();
+                }
+            }
+
+            Files.write(gemfilePath, configuredGemfileContent, StandardCharsets.UTF_8);
+            return gemfilePath.toString();
+        } catch (IOException e) {
+            logger.warn("Error creating/writing to Gemfile {}: {}", gemfilePath, e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Require Bundler setup in the ScriptEngine
+     * 
+     * @param engine
+     */
+    public void bundlerSetup(ScriptEngine engine) {
+        String gemfilePath = getGemfilePath();
+        if (gemfilePath == null) {
+            logger.debug("No Gemfile is found or configured. Skipping Bundler setup.");
+            return;
+        }
+
+        String code = """
+                require 'jruby'
+                JRuby.runtime.instance_config.update_native_env_enabled = false
+                ENV['BUNDLE_GEMFILE'] = '%s'
+                require 'bundler/setup'
```

Comment:
Should we also add `Bundler.require`, so that no action needs to be taken by the user to auto-require gems in the gemfile? This is how bundler/inline behaves. And it can still be "escaped" per-gem in the Gemfile by adding `require: false` to the `gem` line.

Reply:
good idea